### ISSUE TITLE
Expected '!!' and instead saw '?'.

### ISSUE THREAD

**Issue by lennyby93** [douglascrockford/JSLint#193]
I'm not sure what I'm doing wrong, but this piece of code:

        params.rapidCommit = requestIndex.OPTION_RAPID_COMMIT[0] === undefined
            ? true
            : false;

generates this warning:

Expected '!!' and instead saw '?'.
            ? true

In fact even a simple ternary operator generates the same warning.

**Comment by lennyby93** [comment]
Actually this:
        reply.type = params.rapidCommit
            ? "REPLY"
            : "ADVERTISE";

parses fine, if I replace the return values with true/false:
        reply.type = params.rapidCommit
            ? true
            : false;

it triggers the warning.

**Comment by douglascrockford** [comment]
params.rapidCommit = requestIndex.OPTION_RAPID_COMMIT[0] === undefined;

**Comment by lennyby93** [comment]
the issue is with the ternary operator here. Adding ';' in the middle of it is not helping:

        params.rapidCommit = requestIndex.OPTION_RAPID_COMMIT[0] === undefined;
            ? true
            : false;

results in:
JSLint was unable to finish, Unexpected '?'.

**Comment by lennyby93** [comment]
Thx for that, face-palm for missing the obvious. Perhaps message can be made a bit clearer?

**Comment by karimsa** [comment]
The idea behind the warning is you should be using `!!` to convert non-boolean types to boolean types:

```javascript
!!'' // false
!!'javascript' // true
!!undefined // false
!!null // false
```
